frontend/src/components/viewers/MetricsVisualizations.tsx

```diff
@@ -345,40 +345,54 @@ type ConfidenceMetric = {
 };
 
 interface ConfidenceMetricsSectionProps {
-  artifact: Artifact;
+  artifacts: Artifact[];
 }
 
-function ConfidenceMetricsSection({ artifact }: ConfidenceMetricsSectionProps) {
-  const customProperties = artifact.getCustomPropertiesMap();
-  const name = customProperties.get('display_name')?.getStringValue();
+export function ConfidenceMetricsSection({ artifacts }: ConfidenceMetricsSectionProps) {
+  const customPropertiesList = artifacts.map(artifact => artifact.getCustomPropertiesMap());
+  const names = customPropertiesList.map(customProperties =>
+    customProperties.get('display_name')?.getStringValue(),
+  );
 
-  const confidenceMetrics = customProperties
-    .get('confidenceMetrics')
-    ?.getStructValue()
-    ?.toJavaScript();
-  if (confidenceMetrics === undefined) {
+  let confidenceMetricsList = customPropertiesList.map(customProperties =>
+    customProperties
+      .get('confidenceMetrics')
+      ?.getStructValue()
+      ?.toJavaScript(),
+  );
+
+  confidenceMetricsList = confidenceMetricsList.filter(confidenceMetrics => confidenceMetrics);
```

Comment:
Can this line be appended as followed:

```
customPropertiesList.map(customProperties =>
    customProperties
      .get('confidenceMetrics')
      ?.getStructValue()
      ?.toJavaScript(),
  ).filter(confidenceMetrics => confidenceMetrics);
```

Reply:
Sounds good, done: https://github.com/kubeflow/pipelines/pull/8046/commits/0d58f1ef80d36cdd129fa6df92a0b1dcafbd4150